clients/drcachesim/tracer/raw2trace.h

```diff
@@ -827,6 +828,11 @@ template <typename T> class trace_converter_t {
                     buf, (trace_marker_type_t)in_entry->extended.valueB, marker_val);
                 if (in_entry->extended.valueB == TRACE_MARKER_TYPE_KERNEL_EVENT) {
                     impl()->log(4, "Signal/exception between bbs\n");
+                } else if (in_entry->extended.valueB ==
+                           TRACE_MARKER_TYPE_FILTER_ENDPOINT) {
+                    impl()->log(2, "Reached filter endpoint\n");
+                    saw_filter_endpoint = true;
+                    *flush_decode_cache = true;
```

Comment:
I'm trying to understand the reason for this flush.  Flushing the decode cache here is unrelated to filtering and is solely related to windows where code might have changed during the non-traced period?  I don't think that is necessary: the tracer supports modified code and multiple versions of code at the same PC and that should work even with periods of non-tracing.

Reply:
[tagging @abhinav92003 since he had the same question]

I was getting this assert without the cache being flushed. The value for index was greater than instrs.size() which was 1.

```
DEBUG_ASSERT(index >= 0 && index < static_cast<int>(block->instrs.size()));
```
This is related to filtering as far as I can tell. I see this in append_bb_entries, where get_instr_summary is called with instr_count=1, which then calls create_instr_summary which in turn adds an entry to the decode cache.
```
        if (instr_count == 0) {
            // L0 filtering adds a PC entry with a count of 0 prior to each memref.
            skip_icache = true;
            instr_count = 1;
...
...
            instr =
                impl()->get_instr_summary(tls, in_entry->pc.modidx, in_entry->pc.modoffs,
                                          start_pc, instr_count, i, &pc, orig_pc);

```

Reply:
You mean the cache includes block-level info instead of just instr-level info and as such is different for filtered vs non-filtered?  Please add a comment explaining the reason for the flush.

Reply:
Added comment.